### ISSUE TITLE
Bad loss function for classification?

### ISSUE THREAD

**Issue by benmccann** [srendle/libfm#17]
According to the docs:

> For binary classification, cases with y > 0 are regarded as the positive class and with y ≤ 0 as the negative class.

But if you have a target of 0 in the loss function for negative cases then you don't learn anything because your loss is always 0:

```
        } else if (task == 1) {
            grad_loss = target * ( (1.0/(1.0+exp(-target*p))) -  1.0);
```

It seems like target needs to be normalized in the classification case, but I don't see anywhere in the code where that'd be happening. (Note that I haven't actually run the code to prove it's misbehaving, but am just reading it and this didn't make sense to me. Did I miss something?)

**Comment by srendle** [comment]
For classification, the target is normalized to -1,1 in libfm.cpp line 295.
